calculators/election_top_spenders.rb

```diff
@@ -0,0 +1,28 @@
+class ElectionTopSpender
+  def initialize(candidates: [], ballot_measures: [], committees: []); end
+
+  def fetch
+    election_results = ActiveRecord::Base.connection.execute <<~SQL
+      SELECT "election_name", "Tran_NamL", "Tran_NamF", Sum("Tran_Amt1") as "Total_Amount"
+      FROM "combined_contributions"
+      WHERE "election_name" <> ''
+      GROUP BY "election_name", "Tran_NamL", "Tran_NamF";
+    SQL
+
+    results_by_election = election_results.each_with_object({}) do |result, hash|
+      hash[result['election_name']] ||= []
+      hash[result['election_name']] << result
+    end
+
+    top_spenders_by_election = Hash[results_by_election.map do |election_name, spenders|
+      [election_name, spenders.sort_by { |s| s['Total_Amount'] }.reverse.first(3)]
+    end]
+
+    Election.find_each do |election|
+      top_spenders = top_spenders_by_election[election.name]
+      next unless top_spenders
+
+      election.save_calculation(:top_spenders, top_spenders)
```

Comment:
Shouldn't "top spenders" be based on expenditures instead of contributions?  I think 460 Summary Line 11?

Reply:
Don't we want to know who is funding the election not which campaigns spent the most?  You prototype data has people (Bezos) not campaign committees.  If we did not have high independent expenditures then we might see who is giving most to campaigns.  Maybe the name is wrong.  I think maybe we may want to have two stats: biggest contributors to office elections and biggest contributors to ballot measures.